app/services/aws/credentials_service.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module Aws
+  class CredentialsService
+    class << self
+      def call(params = {})
+        new(params).call
+      end
+    end
+
+    attr_reader :params
+
+    ROLE_SESSION_NAME = '3scale-porta'
+
+    def initialize(params)
+      @params = params
+    end
+
+    # @return [Hash] Depending on the available connections, this method may return:
+    # { credentials: Aws::STS::Types::AssumeRoleWithWebIdentityResponse } if STS credentials are available;
+    # { web_identity_token_file: 'foo', role_name: 'bar' } if the values are present;
+    # An empy Hash if no configurations are available;
+    def call
```

Comment:
Hi  @Thales Martinez  @Nidhi Soni  we're hoping to release STS support for RHOAM in the next release, 1.27. To do that we will have the 3scale work around in place i.e. providing an IAM key/secret in an OpenShift secret to 3scale. 
When STS support for 3scale does come out, RHOAM will provide an OpenShift secret with an ARN representing the role to be used for the STS support. 
The question is, will this process be seamless? Can we go from the IAM based secret to the STS based secret without any manual intervention. Will 3scale automatically switch from IAM to STS as long as the correct secret is provided?

Reply:
Hello @briangallagher 👋 
With this implementation, 3Scale will try to use STS credentials if a `web_identity_token_file` file is provided, which **should** be seamless. Now, I can't really guarantee that there won't be any manual intervention for us to get access to this file... Maybe @raelga can help us with this.

Reply:
@thalesmiguel thanks. Can this scenario be tested. The BU are reluctant to roll out STS support for RHOAM is there is complex manual steps required when STS becomes available in 3scale

Reply:
@raelga could you please clarify this?

Reply:
To test the functionality, we a deployed development OCP4.9 cluster with STS:

![Screenshot 2022-09-28 at 17 58 04](https://user-images.githubusercontent.com/795289/192828093-73354ca7-5b65-480b-aea6-b090ce93b09e.png)
https://console-openshift-console.apps.dev-eng-ocp4-sts.dev.3sca.net/k8s/ns/3scale-sts-rael/pods/aws-cli/terminal

The role has been created with `ccoctl` using the follwing `credentialsrequest` to allow S3 access to the test bucket:

```yaml
apiVersion: cloudcredential.openshift.io/v1
kind: CredentialsRequest
metadata:
  name: system-aws-creds
  namespace: openshift-cloud-credential-operator
spec:
  secretRef:
    name: system-aws-creds
    namespace: 3scale-sts-rael
  providerSpec:
    apiVersion: cloudcredential.openshift.io/v1
    kind: AWSProviderSpec
    statementEntries:
      - effect: Allow
        action:
          - "s3:*"
        resource: "arn:aws:s3:::3scale-dev-eng-system-assets"
      - effect: Allow
        action:
          - "s3:*"
        resource: "arn:aws:s3:::3scale-dev-eng-system-assets/*"
  serviceAccountNames:
    - default
    - 3scale-sts-test
 ```

@thalesmiguel  feel free to test porta with the STS feature in the new cluster, the development team has full admin privileges to this cluster using Github authentication. The cluster will be destroyed once the testing is completed.

Reply:
Normally cco is turned to manual mode when the cluster is STS enabled. Is the case that this was tested on non STS enabled cluster? I think it's fine as long as it was possible to confirm that the component given the secret was indeed using it in order to authenticate using STS rather than IAM any IAM credentials.

Reply:
This was tested in a STS enabled cluster with CCO in manual mode.

The `CredentialsRequest`  is not actually deployed in the cluster, it just was used to generate the required role with `ccoctl` in the AWS account and generate the secret that needs to be deployed in the namespace.